src/nfa/nfa.cc

```diff
@@ -59,1097 +78,1159 @@ std::list<Symbol> OnTheFlyAlphabet::get_symbols() const
 
+    return result;
+} // for_begin }}}
 
-bool Vata2::Nfa::is_complete(const Nfa& aut, const Alphabet& alphabet)
+Nfa::const_iterator Nfa::const_iterator::for_end(const Nfa* /* nfa*/)
 { // {{{
-	std::list<Symbol> symbs_ls = alphabet.get_symbols();
-	std::unordered_set<Symbol> symbs(symbs_ls.cbegin(), symbs_ls.cend());
-
-	// TODO: make a general function for traversal over reachable states that can
-	// be shared by other functions?
-	std::list<State> worklist(aut.initialstates.begin(),
-		aut.initialstates.end());
-	std::unordered_set<State> processed(aut.initialstates.begin(),
-		aut.initialstates.end());
+    const_iterator result;
+    result.is_end = true;
+    return result;
+} // for_end }}}
 
-	while (!worklist.empty())
-	{
-		State state = *worklist.begin();
-		worklist.pop_front();
-
-		size_t n = 0;      // counter of symbols
-		for (const auto& symb_stateset : aut[state])
-		{
-			++n;
-			if (!haskey(symbs, symb_stateset.first))
-			{
-				throw std::runtime_error(std::to_string(__func__) +
-					": encountered a symbol that is not in the provided alphabet");
-			}
-
-			const StateSet& stateset = symb_stateset.second;
-			for (const auto& tgt_state : stateset)
-			{
-				bool inserted;
-				tie(std::ignore, inserted) = processed.insert(tgt_state);
-				if (inserted) { worklist.push_back(tgt_state); }
-			}
-		}
-
-		if (symbs.size() != n) { return false; }
-	}
+Nfa::const_iterator& Nfa::const_iterator::operator++()
+{ // {{{
+    assert(nullptr != nfa);
+
+    ++(this->ssIt);
+    const StateSet& state_set = this->tlIt->states_to;
+    assert(!state_set.empty());
+    if (this->ssIt != state_set.end())
+    {
+        this->refresh_trans();
+        return *this;
+    }
+
+    // out of state set
+    ++(this->tlIt);
+    const TransitionList& tlist = this->nfa->get_transitions_from_state(this->trIt);
+    assert(!tlist.empty());
+    if (this->tlIt != tlist.end())
+    {
+        this->ssIt = this->tlIt->states_to.begin();
+
+        this->refresh_trans();
+        return *this;
+    }
+
+    // out of transition list
+    ++this->trIt;
+    assert(!this->nfa->nothing_in_trans());
+    while (this->trIt < this->nfa->transitionrelation.size() &&
+        this->nfa->get_transitions_from_state(this->trIt).empty())
+    {
+        ++this->trIt;
+    }
+
+    if (this->trIt < this->nfa->transitionrelation.size())
+    {
+        this->tlIt = this->nfa->get_transitions_from_state(this->trIt).begin();
+        assert(!this->nfa->get_transitions_from_state(this->trIt).empty());
+        const StateSet& new_state_set = this->tlIt->states_to;
+        assert(!new_state_set.empty());
+        this->ssIt = new_state_set.begin();
+
+        this->refresh_trans();
+        return *this;
+    }
+
+    // out of transitions
+    this->is_end = true;
+
+    return *this;
+} // operator++ }}}
 
-	return true;
-} // is_complete }}}
+//Returns the min_symbol and its post (subset construction), advances the min_symbol to the next minimal symbol,
```

Comment:
Would be better to conform to classical `/**/` comments, so doxygen can generate information from it

Reply:
changed to doxygen format.